src/modules/esl-popup/test/esl-popup-position.behaviour-fit-minor.test.ts

```diff
@@ -0,0 +1,137 @@
+      const expected = Object.assign({}, expectedRef) as PopupPositionValue;
+      expected.arrow = {x: 235, y: 85};
+      expected.popup = popup.shift(260, 290);
+      expect(calcPopupPosition(cfg)).toEqual(expected);
+    });
+
+    test('right when there is a lack of space on the left side in the top position', () => {
+      const cfg = {...cfgRef, position: 'top', outer: container.shift(440, 0)} as PopupPositionConfig;
+      const expected = Object.assign({}, expectedRef) as PopupPositionValue;
+      expected.arrow = {x: 55, y: 85};
+      expected.popup = popup.shift(440, 290);
+      expect(calcPopupPosition(cfg)).toEqual(expected);
+    });
+
+    test('left when there is a lack of space on the right side in the bottom position', () => {
+      const cfg = {...cfgRef, position: 'bottom', outer: container.shift(-440, 0)} as PopupPositionConfig;
+      const expected = Object.assign({}, expectedRef) as PopupPositionValue;
+      expected.arrow = {x: 235, y: 85};
+      expected.popup = popup.shift(260, 530);
+      expected.placedAt = 'bottom';
+      expect(calcPopupPosition(cfg)).toEqual(expected);
+    });
+
+    test('right when there is a lack of space on the left side in the bottom position', () => {
+      const cfg = {...cfgRef, position: 'bottom', outer: container.shift(440, 0)} as PopupPositionConfig;
+      const expected = Object.assign({}, expectedRef) as PopupPositionValue;
+      expected.arrow = {x: 55, y: 85};
+      expected.popup = popup.shift(440, 530);
+      expected.placedAt = 'bottom';
+      expect(calcPopupPosition(cfg)).toEqual(expected);
+    });
+  });
+
+  describe('should not adjust the position in case:', () => {
+    test('when the popup size is greater than the outer limiter size', () => {
+      const narrowContainer = trigger.grow(50);
+      const cfg = {...cfgRef, position: 'top', outer: narrowContainer} as PopupPositionConfig;
+      const expected = Object.assign({}, expectedRef) as PopupPositionValue;
+      expected.popup = popup.shift(360, 290);
+      expect(calcPopupPosition(cfg)).toEqual(expected);
+    });
+
+    describe('when the popup has a position on the vertical axe and', () => {
+      const expected = Object.assign({}, expectedRef) as PopupPositionValue;
+      expected.popup = popup.shift(360, 290);
+      test('when the trigger is outside of the outer limiting element by the own left edge', () => {
+        const cfg = {...cfgRef, position: 'top', outer: container.shift(-510, 0)} as PopupPositionConfig;
+        expect(calcPopupPosition(cfg)).toEqual(expected);
+      });
+
+      test('when the trigger is outside of the outer limiting element by the own right edge', () => {
+        const cfg = {...cfgRef, position: 'top', outer: container.shift(510, 0)} as PopupPositionConfig;
+        expect(calcPopupPosition(cfg)).toEqual(expected);
+      });
+    });
+
+    describe('when the popup has a position on the horizontal axe and', () => {
```

Comment:
```suggestion
    describe('when the popup has a position on the horizontal axis', () => {
```
Same as what was mentioned by @yadamskaya